internal/statemachine/classic_states.go

```diff
@@ -1309,23 +1309,35 @@ func (stateMachine *StateMachine) makeQcow2Img() error {
 	return nil
 }
 
-var updateBootloaderState = stateFunc{"update_bootloader", (*StateMachine).updateBootloader}
+var setupBootloaderState = stateFunc{"setup_bootloader", (*StateMachine).setupBootloader}
 
-// updateBootloader determines the bootloader for each volume
-// and runs the correct helper function to update the bootloader
-func (stateMachine *StateMachine) updateBootloader() error {
+// setupBootloader determines the bootloader for each volume
+// and runs the correct helper function to install/update the bootloader
+func (stateMachine *StateMachine) setupBootloader() error {
 	if stateMachine.RootfsPartNum == -1 || stateMachine.RootfsVolName == "" {
 		return fmt.Errorf("Error: could not determine partition number of the root filesystem")
 	}
+	if stateMachine.BootPartNum == -1 {
+		return fmt.Errorf("Error: could not determine partition number of the boot filesystem")
```

Comment:
Does this mean that now every build will consistently fail if there is no bootloader partition?
As per what we discussed, I think it's *fine* for us not to do the grub-install dance in cases when there's no boot partition *for now*, but I don't think we should fail the build just because of that.

Reply:
I agree and I think I will move this check in the `grub` case below or even not check it at all here and let the `setupGrub` method deal with it based on the kind of setup we want (bios-only/UEFI+BIOS).